SKIRT/core/FluxRecorder.hpp

```diff
@@ -111,21 +111,70 @@ class WavelengthGrid;
     detect() function in thread-local storage. Finally, at the end of the simulation, the
     instrument calls the calibrateAndWrite() function to output the recorded information.
 
+    Flux calibration
+    ----------------
+
+    A FluxRecorder instance handles the conversion of the detected photon packet contributions to a
+    corresponding observed quantity for each spectral and/or spatial bin. This process is called
+    flux calibration and it depends to some extent on whether the instrument associated with the
+    flux recorder is "distant" or "local".
+
+    A distant instrument is placed at a large distance from the model under study. The observed
+    portion of the sky is approximated by a plane perpendicular to the line of sight rather than a
+    part of a sphere. The instrument uses parallel projection and considers all photon packets to
+    originate at the same distance from the instrument. Consequently, the incoming photon packet
+    contributions can simply be accumulated for each bin and flux calibration can occur on the
+    aggregated result. Assuming a non-relativistic distance, the calibration for the flux densities
+    \f$F_\lambda\f$ in an %SED is \f[ F_\lambda = \frac{1}{\Delta\lambda} \, \frac{1}{4\pi
+    d_\mathrm{m}^2} \, L_\mathrm{bin}\f] where \f$L_\mathrm{bin}\f$ is the aggregated photon packet
+    luminosity contribution in a spectral bin, \f$\Delta\lambda\f$ is the width of that bin, and
+    \f$d_\mathrm{m}\f$ is the distance from the model to the instrument. The calibration for the
+    surface brightness values \f$f_\lambda\f$ in an IFU is \f[ f_\lambda = \frac{1}{\Delta\lambda}
+    \, \frac{1}{4\pi d_\mathrm{m}^2} \, \frac{1}{\Omega_\mathrm{distant}}\, L_\mathrm{bin}\f] where
+    \f$\Omega_\mathrm{distant}\f$ is the solid angle subtended by the detector pixel corresponding
+    to the bin. For a distant instrument using parallel projection, the latter is given by
+    \f[\Omega_\mathrm{distant} = 4\arctan\left(\frac{s_\mathrm{x}}{2d_\mathrm{m}}\right)
+    \arctan\left(\frac{s_\mathrm{y}}{2d_\mathrm{m}}\right) \f] where \f$s_\mathrm{x}\f$ and
+    \f$s_\mathrm{y}\f$ represent the field of view (in model units) of a pixel in each direction.
+    For relativistic distances these relations become slightly more involved, as discussed in a
+    seperate concept note on cosmological redshift.
+
+    A local instrument is placed near or even inside the model. The mapping between incoming photon
+    packet trajectories and detector pixels is determined by the instrument. Although all distances
+    are considered to be non-relativistic, photon packets can now originate at various distances
+    from the instrument, with a wide dynamic range. The FluxRecorder class does not support
+    recording %SEDs for local instruments. The calibration for the surface brightness values
```

Comment:
Is there a reason why recording SEDs is not supported? Is the flux in a bin of the SED not the same as the integrated flux over the full image of an IFU with the same wavelength range? It might not make much sense to compute an SED for an all-sky instrument or a perspective instrument, but I don't see any fundamental reasons why it would not work.

Reply:
There is no fundamental reason, other than that it does not seem useful in practice. SEDs are recorded in SKIRT without taking into account any aperture; this seems fine for parallel projected distant instruments but has no real meaning for a perspective instrument, for example. If and when someone comes up with a relevant use case, we can always add the capability.